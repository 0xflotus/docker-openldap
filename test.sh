#!/bin/sh

# Usage
#Â sudo ./test.sh 
# add -v for verbose mode

verbose=$1
error=0
ok=0

echo_start () {
  echo "------- Start $* test -------"
}

echo_error () {
  echo "\n$(tput setaf 1)/!\ $* failed$(tput sgr0)\n"
  error=`expr $error + 1`
}

echo_ok () {
  echo "\n--> $* ok\n"
  ok=`expr $ok + 1`
}

run_test () {
  fction=$1
  out=test/test.out

  echo_start $fction

  if [ -z ${verbose} ]; then
    ./test/$1.sh > $out 2>&1
  else
    ./test/$1.sh | tee $out 2>&1
  fi
  
  if [ "$(grep -c "$2" $out)" -eq 0 ]; then
    echo_error $fction
  else
    echo_ok  $fction
  fi

  rm $out
}


./test/tools/prepare.sh > /dev/null 2>&1
run_test build "Successfully built"
run_test run-simple "dn: dc=example,dc=com"
run_test run-tls "dn: dc=example,dc=com"
./test/tools/end.sh > /dev/null 2>&1

echo "------- Test finished -------"
echo $error " failed " $ok " passed"

