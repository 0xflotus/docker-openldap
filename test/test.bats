#!/usr/bin/env bats
load test_helper

@test "image build" {

  run build_image
  [ "$status" -eq 0 ]

}

@test "ldapsearch new database" {

  run_image -e USE_TLS=false
  wait_service slapd
  run docker exec $CONTAINER_ID ldapsearch -x -h 127.0.0.1 -b dc=example,dc=org
  clear_container

  [ "$status" -eq 0 ]

}

@test "ldapsearch new database with strict TLS" {

  run_image
  wait_service slapd
  run docker exec $CONTAINER_ID ldapsearch -x -h ldap.example.org -b dc=example,dc=org -ZZ
  clear_container

  [ "$status" -eq 0 ]

}

@test "ldapsearch new database with strict TLS and custom ca/crt" {

  run_image -v $BATS_TEST_DIRNAME/ssl:/osixia/slapd/ssl -e SSL_CRT_FILENAME=test-ldap.crt -e SSL_KEY_FILENAME=test-ldap.key -e SSL_CA_CRT_FILENAME=test-ca.crt
  wait_service slapd
  run docker exec $CONTAINER_ID ldapsearch -x -h ldap-test.example.com -b dc=example,dc=org -ZZ
  clear_container

  chown -R $UNAME:$UNAME $BATS_TEST_DIRNAME/ssl || true

  [ "$status" -eq 0 ]

}

@test "ldapsearch existing database and config" {
  skip
  run_image -e USE_TLS=false -v $BATS_TEST_DIRNAME/database:/var/lib/ldap 
  wait_service slapd
  sleep 60
  run docker exec $CONTAINER_ID ldapsearch -x -h 127.0.0.1 -b dc=test-ldap,dc=osixia,dc=net
  clear_container

  chown -R $UNAME:$UNAME $BATS_TEST_DIRNAME/database || true

  [ "$status" -eq 0 ]

}