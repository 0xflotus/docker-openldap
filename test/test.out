docker run --name openldap-repository-test-container --dns=127.0.0.1 -v /home/bgouny/Bureau/pop/docker-openldap/test/tls/ssl:/etc/ldap/ssl -d osixia/openldap:latest  /sbin/my_init --enable-insecure-key
 --> Obtaining IP
 --> 172.17.0.37
ldapsearch -x -h ldap-test.example.com -b dc=example,dc=com -ZZ
# extended LDIF
#
# LDAPv3
# base <dc=example,dc=com> with scope subtree
# filter: (objectclass=*)
# requesting: ALL
#

# example.com
dn: dc=example,dc=com
objectClass: top
objectClass: dcObject
objectClass: organization
o: Example Inc.
dc: example

# admin, example.com
dn: cn=admin,dc=example,dc=com
objectClass: simpleSecurityObject
objectClass: organizationalRole
cn: admin
description: LDAP administrator

# search result
search: 3
result: 0 Success

# numResponses: 3
# numEntries: 2
